redbeat/schedulers.py

```diff
@@ -455,6 +456,9 @@ def maybe_due(self, entry, **kwargs):
         return next_time_to_run
 
     def tick(self, min=min, **kwargs):
+        if self.lock_acquiring_error:
+            logger.debug('beat: Failed to acquire redis lock on init...')
+            raise self.lock_acquiring_error
         if self.lock:
```

Comment:
I've had a quick look over the code and config of redbeat and can't exactly figure out why we just continue if the lock is None here. Is it possible to run redbeat in a mode where locking is disabled? If not we could instead of catching this exception, storing it and rethrowing just potentially throw if the lock is None. However this way we are swallowing the exception that caused the lock not to be set which does not seem ideal!